__tests__/unit/ethereum/gas.test.ts

```diff
@@ -0,0 +1,48 @@
+import { estimateGas } from "../../../libs/ethereum";
+import {
+  TX_GAS_DEFAULT,
+  TX_GAS_MULTIPLIER,
+} from "../../../libs/ethereum/constants";
+
+describe("Estimate gas", () => {
+  const tx = {
+    from: "0xfrom9bF4A217A57A84dfbbA633E008F05378D47d",
+    to: "0xto508a9e024CaCa25f139C6490c0dea49eF37795",
+    value: "10",
+    data: "0xdcc7bceb000000000000000..00499d11e0b6eac",
+  };
+  const mockedWeb3 = {
+    eth: { estimateGas: jest.fn() },
+  };
+
+  test("Should return default * multiplier if web3 call fails", async () => {
+    // Arrange
+    mockedWeb3.eth.estimateGas.mockRejectedValue(
+      new Error("Oops! Web3 failed"),
+    );
+
+    // Act
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const gas = await estimateGas(tx, mockedWeb3);
```

Comment:
We can import `Web3` and change this one to…

```ts
const gas = await estimateGas(tx, (mockedWeb3 as undefined) as Web3);
```

…by doing so, we won't loose type-safety of `tx` parameter and we wont need `// @ts-ignore`.

Reply:
Done! I have a few of these instances so I'll make sure to review these when i tackle #17

Tbh I didn't think this was legit since I'm setting my own function, but I guess it makes sense so thank you for pointing this out :raised_hands: